src/form-validation.ts

```diff
@@ -0,0 +1,108 @@
+import { ValidationEngine } from './validation-engine';
+import {
+  ValidationSchema,
+  RecordValidationSchema,
+  FieldsValidationSchema,
+  FieldValidation,
+  ValidationResult,
+  FormValidationResult,
+} from './model';
+import { isFunction } from './helper';
+
+export class FormValidation {
+  private validationEngine: ValidationEngine;
+
+  constructor(validationSchema: ValidationSchema) {
+    this.validationEngine = new ValidationEngine();
+    this.setupValidationSchema(validationSchema);
+  }
+
+  // TODO: review parameters order, baseFormValidation validateField has fieldId, value, values
+  // the engine validateField has a different order, IMHO it should be always: fieldId, value, values
+  // since values is options.
+  public validateField(
+    fieldId: string,
+    value: any,
+    values?: any
+  ): Promise<ValidationResult> {
+    return this.validationEngine.validateField(values, fieldId, value);
```

Comment:
I agree, we could reorder `validationEngine` params

Reply:
Done on the refactor branch